File: API/ApiRest/models/ticket.js
const mongoose = require("mongoose");

const ticketSchema = mongoose.Schema({
    total:{
        type: Number,
        required: false,
        default: 0.0
    },
    tipo_ticket:{ //Efectivo o Tarjeta
        type:String,
        required: false,
        defafult: "Efectivo"
    },
    id_user_que_abrio:{ 
        type: String,
        required: true,
        default: "Error"
    },
    id_user_que_cerro:{
        type: String,
        required: false,
        default: "Error"
    },
    id_client:{
        type: String,
        required: true,
        default: "Error"
    },
    id_table:{
        type: String,
        required: true,
        default: "Error"
    },
    id_caja:{
        type:String,
        required: false,
        default: "Error"
    },
    name_table:{
        type: String,
        required: true,
        default: "Error"
    },
    comensales:{
        type: Number,
        required: false,
        default: 1
    },
    date:{
        type: Date,
        required: false,
        default: Date.now
    },
    cobrado:{
        type: Boolean,
        required: false,
        default: false,
    }
})
module.exports = mongoose.model('Ticket', ticketSchema);

File: Interfaz_desktop/interfaz_cs/intermodular/api_models/Ticket.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace intermodular
{
    public class Ticket
    {
        public string _id { get; set; }
        public string tipo_ticket { get; set; }
        public string id_user_que_abrio { get; set; }
        public string id_user_que_cerro { get; set; }
        public string id_client { get; set; }
        public string id_table { get; set; }
        public string id_caja { get; set; }
        public string name_table { get; set; }
        public int comensales { get; set; }
        public float total { get; set; }
        public DateTime date { get; set; }
        public Boolean cobrado { get; set; }

        public static Ticket currentTicket;
        public static List<Ticket> openTickets;
        public static List<Ticket> clientTickets;

        public Ticket(int comensales)
        {
            this.comensales = comensales;
            this.id_user_que_abrio = User.currentUser._id;
            this.id_table = Mesa.currentMesa._id;
            this.name_table = Mesa.currentMesa.name;
            this.id_caja = Caja.currentCaja._id;
            this.cobrado = false;
            this.date = DateTime.Now;
            this.id_client = Client.currentClient._id;

        }

        /** NO BORRAR ESTE METODO EN EL MERGE! (PABLO) */
        public static async Task<bool> getClientOpenTickets()
        {
            string url = $"{Staticresources.urlHead}ticket/{Client.currentClient._id}/sin_cobrar";
            HttpResponseMessage httpResponse = await Staticresources.httpClient.GetAsync(url);

            if (httpResponse.IsSuccessStatusCode)
            {
                string content = await httpResponse.Content.ReadAsStringAsync();
                List<Ticket> tickets = JsonSerializer.Deserialize<List<Ticket>>(content);
                if (tickets.Count > 0)
                {
                    openTickets = tickets;
                    return true;
                }
            }
            return false;
        }
    }
}
